source/common/http/filter/fault_filter.cc

```diff
@@ -69,7 +80,11 @@ FaultFilter::~FaultFilter() { ASSERT(!delay_timer_); }
 // if we inject a delay, then we will inject the abort in the delay timer
 // callback.
 FilterHeadersStatus FaultFilter::decodeHeaders(HeaderMap& headers, bool) {
-  if (!matchesTargetCluster()) {
+  if (!matchesTargetUpstreamCluster()) {
+    return FilterHeadersStatus::Continue;
+  }
+
+  if (!matchesDownstreamNodes(headers)) {
     return FilterHeadersStatus::Continue;
```

Comment:
I think I agree with this. @RomanDzhabarov for our use, can't we just use header match and then just add runtime control to whether we match on headers or not? Seems like there should be some compromise here that doesn't do two separate things.